precompiles/xcm/src/lib.rs

```diff
@@ -51,11 +52,20 @@ where
 
         match selector {
             // Dispatchables
-            Action::AssetsWithdraw => Self::assets_withdraw(handle),
+            Action::AssetsWithdrawSS58 => Self::assets_withdraw(handle, BeneficiaryType::SS58),
+            Action::AssetsWithdrawH160 => Self::assets_withdraw(handle, BeneficiaryType::H160),
         }
     }
 }
 
+/// The supported beneficiary account types
+enum BeneficiaryType {
+    /// 256 bit public key is expected
+    SS58,
```

Comment:
Why not AccountId32/AccountId20? SS58 is encoding style, not account type.

Reply:
Sure, you're right.

I added it when I started and forgot to improve.

Reply:
Done, please check.